Frontend/CO.CDP.OrganisationApp/Pages/Organisation/OrganisationInternationalIdentification.cshtml.cs

```diff
@@ -0,0 +1,183 @@
+            if (!validate) return Redirect("/page-not-found");
+        }
+        catch (OrganisationWebApiClient.ApiException<OrganisationWebApiClient.ProblemDetails> aex)
+        {
+            ApiExceptionMapper.MapApiExceptions(aex, ModelState);
+        }
+        catch (ApiException ex) when (ex.StatusCode == 404)
+        {
+            return Redirect("/page-not-found");
+        }
+
+        return Page();
+    }
+
+
+
+    public async Task<IActionResult> OnPost()
+    {
+        var OrganisationIdentificationNumber = OrganisationScheme != null ? RegistrationNumbers!.GetValueOrDefault(OrganisationScheme) : null;
+
+        if (!ModelState.IsValid)
+        {
+            var validate = await ValidateAndSetExistingIdentifiers();
+            if (!validate) return Redirect("/page-not-found");
+
+            return Page();
+        }
+
+        try
+        {
+            var organisation = await organisationClient.GetOrganisationAsync(Id);
+            if (organisation == null) return Redirect("/page-not-found");
+
+            // Create identifiers for OrganisationScheme
+            var identifiers = OrganisationScheme!.Select(scheme => new OrganisationWebApiClient.OrganisationIdentifier(
+                    id: OrganisationIdentificationNumber,
+                    legalName: organisation.Name,
+                    scheme: OrganisationScheme))
+                .ToList();
+
+            await organisationClient.UpdateOrganisationAdditionalIdentifiers(Id, identifiers);
+
+            return RedirectToPage("OrganisationOverview", new { Id });
+        }
+        catch (OrganisationWebApiClient.ApiException<OrganisationWebApiClient.ProblemDetails> aex)
+        {
+            ApiExceptionMapper.MapApiExceptions(aex, ModelState);
+            return Page();
+        }
+        catch (ApiException ex) when (ex.StatusCode == 404)
+        {
+            return Redirect("/page-not-found");
+        }
+    }
+
+    //private async Task<bool> ValidateAndSetExistingIdentifiers()
+    //{
+    //    var organisation = await organisationClient.GetOrganisationAsync(Id);
+    //    if (organisation == null) return false;
+
+    //    var identfiers = organisation.AdditionalIdentifiers;
+    //    identfiers.Add(organisation.Identifier);
+
+    //    ExistingInternationalIdentifiers = identfiers.Select(x => x.Scheme).ToList();
+
+    //    InternationalIdentifiers = await PopulateIdentifierRegistries();
+
+    //    // Check if ExistingInternationalIdentifiers exist in InternationalIdentifiers
+    //    var schemesInRegistries = InternationalIdentifiers.Select(x => x.Scheme).ToHashSet();
+    //    var existingInRegistries = ExistingInternationalIdentifiers.Where(scheme => schemesInRegistries.Contains(scheme) || scheme == Country + "-Other").ToList();
+
+    //    if (schemesInRegistries.Any() && existingInRegistries.Count == 1)
+    //    {
+    //        HasIdentifierToShow = true;
+    //    }
+
+    //    return true;
+    //}
```

Comment:
Probably remove this unless it's coming back imminently

Reply:
done